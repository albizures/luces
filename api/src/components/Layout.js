import React from 'react'
import Head from 'next/head'
import Link from 'next/link'
import { Layout, Menu } from 'antd'

const { Header, Content, Footer } = Layout
const { Item } = Menu

function logout ({ key }) {
  if (key !== '3') {
    return
  }
  document.cookie = 'id_token=; expires=Thu, 18 Dec 2020 12:00:00 UTC'
  document.location.pathname = '/login'
}

export default ({children, withoutMenus}) => (
  <div>
    <Head>
      <link rel='stylesheet' href='https://unpkg.com/antd@3/dist/antd.min.css' />
      <meta name='viewport' content='width=device-width, initial-scale=1' />
    </Head>
    <Layout className='layout'>
      <Header>
        <div className='logo' />
        {
          !withoutMenus && (
            <Menu
              theme='dark'
              mode='horizontal'
              onClick={logout}
              style={{ lineHeight: '64px' }}>
              <Item key='1'>
                <Link href='/'>
                  <a>Cursos</a>
                </Link>
              </Item>
              <Item key='2'>
                <Link href='/categories'>
                  <a>Categorias</a>
                </Link>
              </Item>
              <Item style={{float: 'right'}} key='3'>Salir</Item>
            </Menu>
          )
        }
      </Header>
      <Content style={{ padding: '0 50px' }}>
        <div style={{ background: '#fff', padding: 24, minHeight: 280 }}>
          {children}
        </div>
      </Content>
    </Layout>
    <Footer style={{ textAlign: 'center' }}>
      Luces Beautiful Â©2018 Created by The Digital Catapult
    </Footer>
  </div>
)
