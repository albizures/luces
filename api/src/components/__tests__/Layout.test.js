import React from 'react'
import renderer from 'react-test-renderer'

import Layout from '../Layout.js'

describe('<Layout/>', () => {
  it('with children', () => {
    const component = renderer.create(
      <Layout>
        <div>
          Hello lalal
        </div>
      </Layout>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
