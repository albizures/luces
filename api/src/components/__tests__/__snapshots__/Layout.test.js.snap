// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> with children 1`] = `
<div>
  <div
    className="layout ant-layout"
  >
    <div
      className="ant-layout-header"
    >
      <div
        className="logo"
      />
      <ul
        aria-activedescendant=""
        className="ant-menu ant-menu-dark ant-menu-root ant-menu-horizontal"
        onKeyDown={[Function]}
        role="menu"
        style={
          Object {
            "lineHeight": "64px",
          }
        }
        tabIndex="0"
      >
        <li
          aria-disabled={undefined}
          aria-selected={true}
          className="ant-menu-item ant-menu-item-selected"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
          title={undefined}
        >
          nav 1
        </li>
        <li
          aria-disabled={undefined}
          aria-selected={false}
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
          title={undefined}
        >
          nav 2
        </li>
        <li
          aria-disabled={undefined}
          aria-selected={false}
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
          title={undefined}
        >
          nav 3
        </li>
      </ul>
    </div>
    <div
      className="ant-layout-content"
      style={
        Object {
          "padding": "0 50px",
        }
      }
    >
      <div>
        Hello lalal
      </div>
    </div>
  </div>
  <div
    className="ant-layout-footer"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Ant Design Â©2016 Created by Ant UED
  </div>
</div>
`;
